### ISSUE TITLE
Allow launching a repo directly from a URL

### ISSUE THREAD

**Issue by ibab** [everware/everware#10]
It would be nice if people could add a button to their repo that directly launches it with everware.
We should come up with a way to implement this.

**Comment by anaderi** [comment]
maybe make repo_url an http GET parameter ? 
e.g. `http://everware.xyz?repo_url=https://github.com/everware/everware-dimuon-example`

**Comment by ibab** [comment]
The form on the homepage could then simply make a GET request as well.

I've quickly put together a button we can use: ![bla](https://googledrive.com/host/0B2scijjRTa3HRE52bi00Y2pGcms)

**Comment by betatim** [comment]
+:100: for a button in people's README that let's me launch their analysis!

What happens if we have to do a OAuth detour? Do the parameters surive?

**Comment by ibab** [comment]
Shouldn't be difficult to do. You can park them in a `?next=` query parameter, let the authenticator do its business and send us to our desired destination.

**Comment by betatim** [comment]
Think I figured it out. Aside: we should use the `state` parameter to 'sign' our OAuth requests (and contribute that to @rgbkrk who we took the OAuth stuff from)

Modify `OAuthLoginHandler.get()` like so:

``` python
self.authorize_redirect(
            redirect_uri=redirect_uri,
            client_id=self.authenticator.client_id,
            scope=[],
            response_type='code',
            extra_params={'state': self.create_signed_value('state', '141414')})
```

where `141414` is some value you check for in `GitHubOAuthHandler` with something like so:

``` python
state = self.get_argument('state', '')
assert '141414' == self.get_secure_cookie('state', state)
```

(right where it says "TODO: check if state argument needs checking" :grinning: )

Wondering if we even need to put something in there at all, but presumably it needs some kind of message to 'sign'. So we either put in some string like `yayayay` or the `repo_url` we are saving for later. I think the `*OAuthHandler` doesn't need to know what the message should be, we will know we made the request by the fact that we can decode the signed message. At least that is how I understand it.

I'll make a PR tomorrow morning.

**Comment by betatim** [comment]
First proposal in #15

**Comment by betatim** [comment]
The code from #15 got lost somehow, at least it isn't in the current master.

**Comment by anaderi** [comment]
really? for example look at line 15 of everware/homelander.py
it is the line from the commit

**Comment by cranmer** [comment]
+1 for launch button

**Comment by anaderi** [comment]
fixed by #45. 
the link should be like this:
http://babushk.in:8000/hub/oauth_login?repourl=https://github.com/betatim/everware-demo
